Define the value argument in the TFChainError constructor

Mahmoud-Emad opened the pull request:
### Description

Following the code changes suggested in [this PR discussion](https://github.com/threefoldtech/tfgrid-sdk-ts/pull/3207#discussion_r1705350580), we initially observed that the `Generic Type` handled the failure correctly but the `TFChain` failed. However, later on, @zaelgohary reported a change in the error pattern—neither KeyError nor TFChainError was being generated.

After further debugging with @0oM4R, we identified the root cause: the value of the property was being sent as a `string` or `undefined`, while the method expects an object in the format `{value: 'The actual value'}`. Additionally, we discovered that using `{enumerable: true}` when defining properties helps ensure that the attributes are displayed correctly when errors are thrown.


### Changes

- Send the value of the prop that I want to define as an object that contains a 'value' attr
- Set the 'enumerable: true' to display the error args

### Related Issues

- https://github.com/threefoldtech/tfgrid-sdk-ts/issues/3304

###

![image](https://github.com/user-attachments/assets/1c101246-ca2b-41e6-a706-28c11661bbd8)

### Checklist

- [ ] Tests included
- [x] Build pass
- [ ] Documentation
- [x] Code format and docstrings
- [x] Screenshots/Video attached (needed for UI changes)

Mahmoud-Emad committed:
fix: Define the value arg in the TFChainError constructor:

- Send the value of the prop that i want to define as an object that contain a 'value' attr
- Set the 'enumerable: ture' to display the error args

zaelgohary commented on packages/tfchain_client/src/errors.ts:
```diff
@@ -27,12 +27,28 @@ class TFChainError extends Error {
-    options.method && Object.defineProperty(this, "method", options.method);
-    options.docs && Object.defineProperty(this, "docs", options.docs);
+
+    if (options.keyError) {
+      Object.defineProperty(this, "keyError", { value: options.keyError, enumerable: true });
+    } else {
+      Object.defineProperty(this, "keyError", { value: "GenericError", enumerable: true });
+    }
+
+    if (options.section) {
+      Object.defineProperty(this, "section", { value: options.section, enumerable: true });
+    }
+
+    if (options.args) {
+      Object.defineProperty(this, "args", { value: options.args, enumerable: true });
+    }
+
+    if (options.method) {
+      Object.defineProperty(this, "method", { value: options.method, enumerable: true });
+    }
+
+    if (options.docs) {
+      Object.defineProperty(this, "docs", { value: options.docs, enumerable: true });
+    }
```
Instead of the repitition, can we do sth like:

```suggestion
    const { keyError = "GenericError", section, args, method, docs } = options;

    Object.defineProperties(this, {
      keyError: { value: keyError, enumerable: true },
      section: { value: section, enumerable: true },
      args: { value: args, enumerable: true },
      method: { value: method, enumerable: true },
      docs: { value: docs, enumerable: true },
    });
```

zaelgohary replied:
Will discard the comment as we need to define properties only if its correspondent option is defined.

AhmedHanafy725 replied:
I didn't test it but maybe something like that
```ts
    if (!options.keyError) {
      options.keyError = "GenericError"
    }
    for (const key of ["keyError", "section", "args", "method", "docs"]) {
      if (options[key]) {
        Object.defineProperty(this, key, { value: options[key], enumerable: true });
      }
```

Mahmoud-Emad replied:
Pretty well. I also think that defining the key dynamically would enhance the code's scalability for future use.

```ts
class TFChainError extends Error {
  constructor(options: ITFChainError) {
    super(options.message);
    this.name = "TFChainError";

    if (!options.keyError) {
      options.keyError = "GenericError";
    }

    Object.keys(options).forEach(key => {
      if (options[key as keyof ITFChainError]) {
        Object.defineProperty(this, key, {
          value: options[key as keyof ITFChainError],
          enumerable: true,
        });
      }
    });
  }
}
```

Mahmoud-Emad committed:
fix: dynamically defines the TFChainError key prop:

- dynamically defines the TFChainError key property instead of making a condition on each key in the 'TFChainError' contractor.
